src/command/dev/next/router/binary.rs

```diff
@@ -23,6 +24,46 @@ pub enum RouterLog {
     Stderr(String),
 }
 
+impl fmt::Display for RouterLog {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let warn_prefix = Style::WarningPrefix.paint("WARN:");
+        let error_prefix = Style::ErrorPrefix.paint("ERROR:");
+        let unknown_prefix = Style::ErrorPrefix.paint("UNKNOWN:");
+        match self {
+            Self::Stdout(stdout) => {
+                if let Ok(parsed) = serde_json::from_str::<serde_json::Value>(stdout) {
+                    let fields = &parsed["fields"];
+                    let level = parsed["level"].as_str().unwrap_or("UNKNOWN");
+                    let message = fields["message"]
+                        .as_str()
+                        .or_else(|| {
+                            // Message is in a slightly different location depending on the
+                            // version of Router
+                            parsed["message"].as_str()
+                        })
+                        .unwrap_or(stdout);
+
+                    match level {
+                        "INFO" => tracing::info!(%message),
+                        "DEBUG" => tracing::debug!(%message),
+                        "TRACE" => tracing::trace!(%message),
+                        "WARN" => write!(f, "{} {}", warn_prefix, &message)?,
+                        "ERROR" => write!(f, "{} {}", error_prefix, &message)?,
+                        "UNKNOWN" => write!(f, "{} {}", unknown_prefix, &message)?,
+                        _ => {}
```

Comment:
I feel like there's some sort of due diligence I could suggest to make sure the level we're setting is right (based on what's in the router, which I've sure you've looked at), but it's sort of escaping me right now; the only thing that sort of worries me is the catch-all at the bottom not emitting _anything_ (but maybe it's better to emit nothing than to emit a bunch of "warning! no level set!" or whatever)

Reply:
Even though this is the same as [the original implementation](https://github.com/apollographql/rover/blob/main/src/command/dev/legacy/router/runner.rs#L222)
Maybe this should be `_ => write!(f, "{} {}", unknown_prefix, &message)?` as well?